ext/include/opentelemetry/ext/zpages/tracez_data_aggregator.h

```diff
@@ -0,0 +1,202 @@
+#pragma once
+
+#include <array>
+#include <atomic>
+#include <condition_variable>
+#include <list>
+#include <map>
+#include <mutex>
+#include <string>
+#include <thread>
+#include <unordered_set>
+#include <vector>
+
+#include "opentelemetry/ext/zpages/latency_boundaries.h"
+#include "opentelemetry/ext/zpages/tracez_processor.h"
+#include "opentelemetry/nostd/string_view.h"
+#include "opentelemetry/sdk/trace/span_data.h"
+#include "opentelemetry/trace/canonical_code.h"
+
+using opentelemetry::sdk::trace::SpanData;
+using opentelemetry::trace::CanonicalCode;
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace ext {
+namespace zpages {
+
+/**
+ * kMaxNumberOfSampleSpans is the maximum number of running, completed or error
+ * sample spans stored at any given time for a given span name.
+ * This limit is introduced to reduce memory usage by trimming sample spans
+ * stored.
+ */
+const int kMaxNumberOfSampleSpans = 5;
+
+/**
+ * TracezSpanData is the span data to be displayed for tracez zpages that is
+ * stored for each span name.
+ */
+struct TracezSpanData {
+  /**
+   * TODO: At this time the maximum count is unknown but a larger data type
+   * might have to be used in the future to store these counts to avoid overflow
```

Comment:
I am not sure what the maximum limit is but I just left that comment in there if there were a lot a spans and there were errors thrown this would be a good place to look. It's possible int64 might also not be enough or is there a known maximum number of spans?